File: gigster/gtest-task1.js
/*
  The entrance fee of the car parking lot is 2;
  The first full or partial hour costs 3;
  Each successive full or partial hour (after the first) costs 4.
  Times are the same day and E < L
*/

'use strict';
function solution(E, L) {
  var HOUR_COST = 4;
  var cost = 5; //entrance fee + first hour;

  //leave - enter.  convert to DateTime and get time span
  var duration = (new Date("1/1/2015 " + L) - new Date("1/1/2015 " + E)) / 60000; // convert to minutes

  if (duration > 60) {
    duration -= 60;
    cost += (Math.floor(duration / 60) * HOUR_COST);
    if (duration % 60) {
      cost += HOUR_COST;
    }
  }
  return cost;
}


console.log(solution("10:00", "10:21"));
console.log(solution("10:59", "11:01"));
console.log(solution("09:42", "11:42"));
console.log(solution("10:00", "13:21"));

File: gigster/gtest-task1.rb
require "byebug"

def solution(e, l)
  # write your code in Ruby 2.2
  # entrance fee + first hour = 5
  # need to convert e and l into date strings or minutes

  #assumptions
  # you only park the car in the parking lot for last than 24 hours
  # the "HH:MM" strings are in military time meaning "00:00" to "24:00"

  hour_cost = 4
  cost = 5 #entrance and first hour

  enter_time = convert_to_min(e)
  exit_time = convert_to_min(l)
  duration = exit_time - enter_time

  if duration > 60
    duration -= 60
    cost += (duration / 60) * hour_cost
    if duration % 60 != 0
      cost += hour_cost
    end
  end
  cost
end

def convert_to_min(time_string)
  arr = time_string.split(":")
  minutes = (arr[0].to_i * 60) + arr[1].to_i
end
